tests/test_url.py

```diff
@@ -501,6 +501,21 @@ def test_with_path_fragment():
     assert str(url.with_path('/test')) == 'http://example.com/test'
 
 
+def test_with_path_empty():
+    url = URL('http://example.com/test')
+    assert str(url.with_path('')) == 'http://example.com'
+
+
+def test_with_path_slash():
+    url = URL('http://example.com/test')
+    assert str(url.with_path('/')) == 'http://example.com/'
```

Comment:
I didn't know about this part of the functionality.
But I agree with you.

I will remove this test.
So slash remaining can be realized as a separate issue.